eng/pipelines/templates/stages/archetype-sdk-tool-python.yml

```diff
@@ -0,0 +1,126 @@
+                parameters:
+                  versionSpec: '${{ parameters.PythonVersion }}'
+
+              - script: |
+                  python --version
+                  python -m pip install virtualenv setuptools wheel packaging build
+                displayName: 'Setup Python Environment'
+
+              - ${{ if eq(parameters.BuildUsingPyProjectToml, true) }}:
+                - pwsh: |
+                    Write-Host "Building python tool artifacts using pyproject.toml"
+                    python -m build  --outdir $(Build.ArtifactStagingDirectory)
+                  displayName: 'Build ${{ parameters.PackageName }}'
+                  workingDirectory: $(Build.SourcesDirectory)/${{ parameters.PackagePath }}
+
+              - ${{ else }}:
+                - pwsh: |
+                    Write-Host "Building python tool artifacts using setup.py"
+                    python setup.py bdist_wheel --universal --dist-dir $(Build.ArtifactStagingDirectory)
+                  displayName: 'Build ${{ parameters.PackageName }}'
+                  workingDirectory: $(Build.SourcesDirectory)/${{ parameters.PackagePath }}
+
+              - task: 1ES.PublishPipelineArtifact@1
+                inputs:
+                  targetPath: '$(Build.ArtifactStagingDirectory)'
+                  artifactName: ${{ parameters.ArtifactName }}
+
+          - ${{ if not(eq(length(parameters.TestSteps), 0)) }}:
+            - job: 'Test'
+
+              dependsOn:
+                - 'Build'
+
+              pool:
+                name: $(LINUXPOOL)
+                image: $(LINUXNEXTVMIMAGE)
+                os: linux
+
+              steps:
+                - template: /eng/pipelines/templates/steps/use-python-version.yml
+                  parameters:
+                    versionSpec: '${{ parameters.PythonVersion }}'
+
+                - ${{ parameters.TestSteps }}
+
+      - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
+        - stage: 'Release'
+          dependsOn: Build
+          condition: Succeeded()
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+            - template: /eng/pipelines/templates/variables/image.yml
+          jobs:
+          - job: PublishPackage
+            displayName: 'Publish ${{ parameters.PackageName }} package to devops feed'
+            pool:
+              name: $(LINUXPOOL)
+              image: $(LINUXNEXTVMIMAGE)
+              os: linux
+            steps:
+            - checkout: none
+            - download: current
+            - task: UsePythonVersion@0
+
+            - script: |
+                set -e
+                python -m pip install twine
+              displayName: Install Twine
+
+            - task: TwineAuthenticate@0
+              displayName: 'Twine Authenticate to feed'
+              inputs:
+                artifactFeeds: ${{ parameters.FeedName }}
+
+            - task: PipAuthenticate@1
```

Comment:
Do we not have a shared template for publishing python packages? Perhaps this template can start to become that.